client/src/views/Chat.vue

```diff
@@ -196,18 +196,40 @@ export default {
     },
 
     switchTo: async function (id) {
-      // switch to the user select
-      this.to = id;
-      await this.retrieve(id);
-      const publicKey = this.$cookies.get(this.to);
-      this[this.to] = Buffer.from(publicKey, "base64").toString();
-      this.publicKey = forge.pki.publicKeyFromPem(this[this.to]);
+    // Check the publicKey first
+    const publicKey = this.$cookies.get(id);
+    if (!publicKey || typeof publicKey !== 'string') {
+        console.error('Invalid publicKey for user:', id);
+        this.toastmsg = "Error: Invalid chat session.";
+        this.toastType = "error";
+        setTimeout(() => {
+            this.toastmsg = "";
+            this.toastType = "";
+        }, 2000);
+        return;
```

Comment:
Lets connect to the user again, using the `connect` method. If the public key is still invalid just disable the chat by disabling the `MessageForm`, the use can view existing messages but not send a new. Show the message `user is inactive` in the chat window.

WDYT?

Reply:
Just pushed some new changes, let me know if this could be ok! 😉